pkg/rpcmiddleware/discover/discover.go

```diff
@@ -74,11 +74,44 @@ func WithTransportCredentials(creds credentials.TransportCredentials) []grpc.Dia
 	}
 }
 
-// WithInsecure returns a DialOption which disables transport security and discovers the default insecure listen port if
-// not specified in the dial target.
+// WithInsecure returns gRPC dial options which disable transport security and discovers the default insecure listen
+// port if not specified in the dial target.
 func WithInsecure() []grpc.DialOption {
 	return []grpc.DialOption{
 		grpc.WithInsecure(),
 		grpc.WithContextDialer(resolver(false)),
 	}
 }
+
+type tlsFallbackKeyType struct{}
+
+var tlsFallbackKey tlsFallbackKeyType
```

Comment:
Why not just name the type `tlsFallbackKey` and use `tlsFallbackKey{}` directly?

Reply:
Because that's a different instance. See also https://golang.org/src/context/context.go#L138